### PR TITLE
Helm: Avoid unneccesary builds

### PR DESCRIPTION
**Pull request by thsig**
**Avoid unnecessary Helm builds**

Previously, Helm modules would get rebuilt ahead of deployment, even if the service version hadn't changed (thus leading deploy tasks to not redeploy the service, making the build step unnecessary).

We address that here by making the Helm build handler a no-op, and instead running its logic inside the deployment handler.

### CODE REVIEW

**Review by eysi09 — DISMISSED** [review]

**Comment by edvald** [comment]
Wanna revive this one @thsig?

File: core/test/integ/src/plugins/kubernetes/helm/common.ts [hunk 1]
```diff
@@ -46,26 +48,30 @@ export async function getHelmTestGarden() {
...
       })
   )
-  const results = await garden.processTasks(tasks)
-
-  const err = first(Object.values(results).map((r) => r && r.error))
-
-  if (err) {
-    throw err
-  }
+  await garden.processTasks(buildTasks)
```
Comment (edvald) [inline core/test/integ/src/plugins/kubernetes/helm/common.ts:71]: May not really matter here but FYI, you can make this `await garden.processTasks(buildTasks, { throwOnError: true })` to make sure errors are thrown.

File: core/src/plugins/kubernetes/helm/deployment.ts [hunk 2]
```diff
@@ -211,3 +227,108 @@ export async function deleteService(params: DeleteServiceParams): Promise<HelmSe
...
+        log.debug("Fetching chart (after updating)...")
+        await pullChart(k8sCtx, log, module)
+      }
+    }
+  })
+}
+
+async function pullChart(ctx: KubernetesPluginContext, log: LogEntry, module: HelmModule) {
+  if (await chartsFetched(log, module)) {
+    log.debug(`Charts already fetched for module ${module.name}`)
```
Comment (edvald) [inline core/src/plugins/kubernetes/helm/deployment.ts:273]: _Super_ minor nitpick, but it's always a single chart, not plural.

**Review by edvald — APPROVED** [review]
LGTM 👍 Very minor comments, barely worth actioning

**Comment by edvald** [comment]
@thsig Tests seem to be legitimately failing. Please make sure those are okay before merging.

**Pull request closed without merging**